Documentation for new QML library (#726)
MicrosoftDocs/quantum-docs-pr: articles/libraries/machine-learning/design.md

---

> title: Quantum machine learning library

Reviewer: Make title accurately reflect content

description: Learn the basic concepts of designing circuit models for the quantum circuit centric classifier.
author: geduardo
ms.author: v-edsanc@microsoft.com
ms.date: 02/17/2020
ms.topic: article
uid: microsoft.quantum.libraries.machine-learning.design
---

# Design your own classifier

> In this tutorial you will learn the basic concepts behind the design of circuit models for the quantum circuit centric classifier.

Author: ```suggestion
In this tutorial you, will learn the basic concepts behind the design of circuit models for the quantum circuit centric classifier.
```

> In this tutorial you will learn the basic concepts behind the design of circuit models for the quantum circuit centric classifier.

Reviewer: ```suggestion
In this tutorial you will learn how to design your own quantum circuit centric classifier.
```

> In this tutorial you will learn the basic concepts behind the design of circuit models for the quantum circuit centric classifier.

Reviewer: Or similar. A tutorial is meant to be hands on

Reviewer: Is this really a how to guide?

Reviewer: What is the intent of this article?

Reviewer 2: Rewording to "guide"

- A large number of parameters implies a more flexible model, which may be suitable to draw complicated classification boundaries but which may also be more susceptible to overfitting.

- Entangling gates between qubits are essential to capture the correlations between the quantum features.

## How to build a classifier with Q\#

To build a classifier we are going to concatenate parametrized controlled rotations in our circuit model. To do it we can use the type [`ControlledRotation`](xref:microsoft.quantum.machinelearning.controlledrotation) defined in the Quantum Machine Learning library. This type accepts four arguments that determine: the index of the target qubit, the array of indices of the control qubits, the axis of rotation, and index of the associated parameter in the array of parameters defining the model.

Let's see an example of a classifier. In the [half-moons sample](link to the sample), we can find the following classifier defined in the file `Training.qs`.

>     ```qsharp

Reviewer: Code isn't rendering properly

Author: Looks like it's indented too far, perhaps?
